backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/service/SpecificationDataParser.kt

```diff
@@ -96,7 +96,8 @@ class SpecificationDataParser @Autowired constructor(
                                   metadata: SpecificationFileMetadata? = null
     ): SpecificationMetadata {
         val idFromUpload = extractId(fileContent)
-        val serviceId = getConsistentServiceId(idFromUpload, idFromPath)
+        val serviceId = getConsistentServiceId(listOf(idFromUpload, idFromPath, metadata?.id))
```

Comment:
So, either that should be a `vararg` where you can put any amount of ids in (and the order doesn't matter), or it should be a specific type like `Triple`.